### ISSUE TITLE
Improve protobuf detection

### ISSUE THREAD

**Issue by ktf** [FairRootGroup/FairRoot#645]
At the moment the only way to correctly specify protobuf is to do something along the lines of:

```
  -DPROTOBUF_INCLUDE_DIR=$(dirname $(which protoc))/../include                \
  -DPROTOBUF_PROTOC_EXECUTABLE=$(which protoc)                                \
  -DPROTOBUF_LIBRARY=$(dirname $(which protoc))/../lib/libprotobuf.$SONAME    \
```

This is because at the moment the lookup strategy excludes system locations and the above is the only way to have something which works both in the case of our own protobuf and for the system one. Would it be possible to change things so that:

- If `ProtoBuf_DIR` is defined, protobuf is looked up there (and the build fails if not found).
- If `ProtoBuf_DIR` is not defined, something like the above logic is used to determine the system protobuf.

**Comment by dennisklein** [comment]
#632 will address this issues. I enables lookup in system paths and if `PROTOBUF_ROOT` is defined, it searches there first.

After the next FairRoot release (this month) we will merge #632 into `dev`. Can this wait until then or do you need it faster? Btw, #632 will change the logic for all dependencies as described above.

**Comment by MohammadAlTurany** [comment]
Hi Giulio,
Could you check if the PR #646 solve this issue

**Comment by dennisklein** [comment]
The upstream FindProtobuf.cmake does not offer a hint variable, that's why it was copied in the past and why it is synched with upstream in #632.

**Comment by MohammadAlTurany** [comment]
solved in  #652
